src/Utilities/CssDocument.php

```diff
@@ -82,23 +61,47 @@ public function __construct(string $css)
      */
     public function getStyleRulesData(array $allowedMediaTypes): array
     {
-        $splitCss = $this->splitCssAndMediaQuery($allowedMediaTypes);
-
         $ruleMatches = [];
-        foreach ($splitCss as $cssPart) {
-            // process each part for selectors and definitions
-            \preg_match_all('/(?:^|[\\s^{}]*)([^{]+){([^}]*)}/mi', $cssPart['css'], $matches, PREG_SET_ORDER);
-
-            foreach ($matches as $cssRule) {
+        /** @var CssRenderable $rule */
+        foreach ($this->sabberwormCssDocument->getContents() as $rule) {
+            if ($rule instanceof CssAtRuleBlockList && $rule->atRuleName() === 'media') {
+                /** @var string $mediaQueryList */
```

Comment:
I propose we move this code block into a private method.

Reply:
Will do.  Just looked to see if it would be possible to create a separate method for this as a separate PR (as the functionality is pre-existing), but it seems not - the pre-existing code is tangled up with a `preg_split`, and I don't see much point untangling it if it's going to be replaced anyway.

Reply:
Done.